--- src/src/fdevent.c.orig
+++ src/src/fdevent.c
@@ -433,9 +433,6 @@
 /* iOS does not allow subprocess creation; avoid compiling advanced interfaces*/
 #if defined(__APPLE__) && defined(__MACH__)
 #include <TargetConditionals.h> /* TARGET_OS_IPHONE, TARGET_OS_MAC */
-#if TARGET_OS_IPHONE            /* iOS, tvOS, or watchOS device */
-#undef HAVE_POSIX_SPAWN
-#endif
 #endif
 
 
@@ -656,7 +653,7 @@
 
     int errnum = errno;
     int argnum =
-      (0 == strcmp(argv[0], "/bin/sh") && argv[1] && 0 == strcmp(argv[1], "-c"))
+      (0 == strcmp(argv[0], "/var/usr/bin/sh") && argv[1] && 0 == strcmp(argv[1], "-c"))
       ? 2
       : 0;
     perror(argv[argnum]);
--- src/src/gw_backend.c.orig
+++ src/src/gw_backend.c
@@ -1677,8 +1677,8 @@
 
                     host->args.ptr = ck_calloc(4, sizeof(char *));
                     host->args.used = 3;
-                    host->args.ptr[0] = ck_malloc(sizeof("/bin/sh"));
-                    memcpy(host->args.ptr[0], "/bin/sh", sizeof("/bin/sh"));
+                    host->args.ptr[0] = ck_malloc(sizeof("/var/usr/bin/sh"));
+                    memcpy(host->args.ptr[0], "/var/usr/bin/sh", sizeof("/var/usr/bin/sh"));
                     host->args.ptr[1] = ck_malloc(sizeof("-c"));
                     memcpy(host->args.ptr[1], "-c", sizeof("-c"));
                     host->args.ptr[2] = ck_malloc(sizeof("exec ")-1
