--- src/deps/v8/src/base/platform/semaphore.h.orig
+++ src/deps/v8/src/base/platform/semaphore.h
@@ -11,7 +11,7 @@
 #include "src/base/win32-headers.h"
 #endif
 
-#if V8_OS_DARWIN
+#if 0
 #include <dispatch/dispatch.h>
 #elif V8_OS_ZOS
 #include "zos-semaphore.h"
@@ -57,7 +57,7 @@
   // the semaphore counter is decremented and true is returned.
   bool WaitFor(const TimeDelta& rel_time) V8_WARN_UNUSED_RESULT;
 
-#if V8_OS_DARWIN
+#if 0
   using NativeHandle = dispatch_semaphore_t;
 #elif V8_OS_POSIX
   using NativeHandle = sem_t;
--- src/deps/v8/src/base/platform/semaphore.cc.orig
+++ src/deps/v8/src/base/platform/semaphore.cc
@@ -4,7 +4,7 @@
 
 #include "src/base/platform/semaphore.h"
 
-#if V8_OS_DARWIN
+#if 0
 #include <dispatch/dispatch.h>
 #elif V8_OS_WIN
 #include <windows.h>
@@ -19,7 +19,7 @@
 namespace v8 {
 namespace base {
 
-#if V8_OS_DARWIN
+#if 0
 
 Semaphore::Semaphore(int count) {
   native_handle_ = dispatch_semaphore_create(count);
