--- source/mono/eglib/glib.h.orig
+++ source/mono/eglib/glib.h
@@ -1487,6 +1487,10 @@
 #define g_try_realloc(obj, size) (g_cast (monoeg_try_realloc ((obj), (size))))
 #define g_memdup(mem, size) (g_cast (monoeg_g_memdup ((mem), (size))))
 
+#if defined(__MACH__)
+typedef int clockid_t;
+#endif
+
 #ifndef G_OS_WIN32
 gint
 g_clock_nanosleep (clockid_t clockid, gint flags, const struct timespec *request, struct timespec *remain);
