--- src/crypto/async/arch/async_posix.h.orig
+++ src/crypto/async/arch/async_posix.h
@@ -19,7 +19,8 @@
 #include <unistd.h>
 
 #if _POSIX_VERSION >= 200112L \
-    && (_POSIX_VERSION < 200809L || defined(__GLIBC__) || defined(__FreeBSD__))
+    && (_POSIX_VERSION < 200809L || defined(__GLIBC__) || defined(__FreeBSD__)) && \
+    !(defined(__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__) && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 1050)
 
 #include <pthread.h>
 
