--- src/libcxx/src/filesystem/operations.cpp.orig
+++ src/libcxx/src/filesystem/operations.cpp
@@ -764,9 +764,16 @@
 
 uintmax_t remove_all_impl(int parent_directory, const path& p, error_code& ec) {
   // First, try to open the path as a directory.
+#ifdef O_CLOEXEC
   const int options = O_CLOEXEC | O_RDONLY | O_DIRECTORY | O_NOFOLLOW;
+#else
+  const int options = O_RDONLY | O_DIRECTORY | O_NOFOLLOW;
+#endif
   int fd            = ::openat(parent_directory, p.c_str(), options);
   if (fd != -1) {
+#ifndef O_CLOEXEC
+    ::fcntl(fd, F_SETFD, FD_CLOEXEC);
+#endif
     // If that worked, iterate over the contents of the directory and
     // remove everything in it, recursively.
     DIR* stream = ::fdopendir(fd);
