--- src/libcxx/include/__mbstate_t.h.orig
+++ src/libcxx/include/__mbstate_t.h
@@ -42,6 +42,7 @@
 #elif __has_include(<bits/types/mbstate_t.h>)
 #  include <bits/types/mbstate_t.h> // works on most Unixes
 #elif __has_include(<sys/_types/_mbstate_t.h>)
+#  include <machine/types.h>
 #  include <sys/_types/_mbstate_t.h> // works on Darwin
 #elif _LIBCPP_HAS_WIDE_CHARACTERS && __has_include_next(<wchar.h>)
 #  include_next <wchar.h> // fall back to the C standard provider of mbstate_t
