--- src/libcxx/CMakeLists.txt.orig
+++ src/libcxx/CMakeLists.txt
@@ -17,6 +17,7 @@
 
 set(CMAKE_FOLDER "libc++")
 
+set(CMAKE_SHARED_LINKER_FLAGS "-Wl,-reexport_library,lib/libc++abi.1.0.dylib")
 set(LIBCXX_SOURCE_DIR  ${CMAKE_CURRENT_SOURCE_DIR})
 set(LIBCXX_BINARY_DIR  ${CMAKE_CURRENT_BINARY_DIR})
 
--- src/libcxxabi/src/CMakeLists.txt.orig
+++ src/libcxxabi/src/CMakeLists.txt
@@ -238,7 +238,7 @@
 function(reexport_symbols file)
   export_symbols("${file}")
   # -reexported_symbols_list is only available on Apple platforms
-  if (APPLE)
+  if (0)
     target_link_libraries(cxxabi-reexports INTERFACE "-Wl,-reexported_symbols_list,${file}")
   endif()
 endfunction()
