--- src/libcxx/src/atomic.cpp.orig
+++ src/libcxx/src/atomic.cpp
@@ -29,6 +29,15 @@
        __ENVIRONMENT_WATCH_OS_VERSION_MIN_REQUIRED__ >= 100400)
 #    define _LIBCPP_USE_OS_SYNC
 #  endif
+#  if (defined(__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__) &&                                                       \
+       __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 101500) ||                                                     \
+      (defined(__ENVIRONMENT_IPHONE_OS_VERSION_MIN_REQUIRED__) &&                                                      \
+       __ENVIRONMENT_IPHONE_OS_VERSION_MIN_REQUIRED__ >= 130000) ||                                                    \
+      (defined(__ENVIRONMENT_TV_OS_VERSION_MIN_REQUIRED__) && __ENVIRONMENT_TV_OS_VERSION_MIN_REQUIRED__ >= 130000) || \
+      (defined(__ENVIRONMENT_WATCH_OS_VERSION_MIN_REQUIRED__) &&                                                       \
+       __ENVIRONMENT_WATCH_OS_VERSION_MIN_REQUIRED__ >= 60000)
+#    define _LIBCPP_HAS_ULOCK
+#  endif
 #endif
 
 #ifdef __linux__
@@ -113,7 +122,7 @@
   _LIBCPP_FUTEX(__ptr, FUTEX_WAKE_PRIVATE, __notify_one ? 1 : INT_MAX, 0, 0, 0);
 }
 
-#elif defined(__APPLE__)
+#elif defined(__APPLE__) && defined(_LIBCPP_HAS_ULOCK)
 
 #  if defined(_LIBCPP_USE_OS_SYNC)
 
