--- src/libcxxabi/src/cxa_exception.cpp.orig
+++ src/libcxxabi/src/cxa_exception.cpp
@@ -515,6 +515,7 @@
 * If it has been rethrown, there is nothing to do.
 * Otherwise delete the exception and pop the catch stack to empty.
 */
+extern "C" void _Unwind_DeleteException(_Unwind_Exception *);
 void __cxa_end_catch() {
   static_assert(sizeof(__cxa_exception) == sizeof(__cxa_dependent_exception),
                 "sizeof(__cxa_exception) must be equal to "
