diff -urN a/apt-pkg/CMakeLists.txt b/apt-pkg/CMakeLists.txt
--- a/apt-pkg/CMakeLists.txt
+++ b/apt-pkg/CMakeLists.txt
@@ -77,7 +77,6 @@
 set_target_properties(apt-pkg PROPERTIES VERSION ${MAJOR}.${MINOR})
 set_target_properties(apt-pkg PROPERTIES SOVERSION ${MAJOR})
 set_target_properties(apt-pkg PROPERTIES CXX_VISIBILITY_PRESET hidden)
-target_link_libraries(apt-pkg PRIVATE -Wl,-version-script="${CMAKE_CURRENT_BINARY_DIR}/versionscript")
 
 # Install the library and the header files
 install(TARGETS apt-pkg LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
diff -urN a/apt-pkg/contrib/fileutl.cc b/apt-pkg/contrib/fileutl.cc
--- a/apt-pkg/contrib/fileutl.cc
+++ b/apt-pkg/contrib/fileutl.cc
@@ -1869,12 +1869,12 @@
 	 // Drain compressed buffer as far as possible.
 	 ZSTD_inBuffer in = {
 	    .src = zstd_buffer.get(),
-	    .size = zstd_buffer.size(),
+	    .size = static_cast<size_t>(zstd_buffer.size()),
 	    .pos = 0,
 	 };
 	 ZSTD_outBuffer out = {
 	    .dst = To,
-	    .size = Size,
+	    .size = static_cast<size_t>(Size),
 	    .pos = 0,
 	 };
 
@@ -1912,7 +1912,7 @@
       };
       ZSTD_inBuffer in = {
 	 .src = From,
-	 .size = Size,
+	 .size = static_cast<size_t>(Size),
 	 .pos = 0,
       };
 
diff -urN a/apt-pkg/deb/dpkgpm.cc b/apt-pkg/deb/dpkgpm.cc
--- a/apt-pkg/deb/dpkgpm.cc
+++ b/apt-pkg/deb/dpkgpm.cc
@@ -1772,8 +1772,6 @@
    bool dpkgMultiArch = _system->MultiArchSupported();
    bool dpkgProtectedField = debSystem::AssertFeature("protected-field");
 
-   // start pty magic before the loop
-   StartPtyMagic();
 
    // Tell the progress that its starting and fork dpkg
    d->progress->Start(d->master);
@@ -2166,7 +2164,6 @@
       }
    }
    // dpkg is done at this point
-   StopPtyMagic();
    CloseLog();
 
    if (d->dpkg_error.empty() == false)
diff -urN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,10 +27,6 @@
 set(CMAKE_EXE_LINKER_FLAGS_COVERAGE "-lgcov")
 set(CMAKE_SHARED_LINKER_FLAGS_COVERAGE "-lgcov")
 
-# Work around bug in GNUInstallDirs
-if (EXISTS "/etc/debian_version")
-  set(CMAKE_INSTALL_LIBEXECDIR "lib")
-endif()
 
 # Include stuff
 include(Misc)
@@ -87,6 +83,7 @@
 add_optional_compile_options(Werror=suggest-override)
 add_optional_compile_options(Werror=return-type)
 add_optional_compile_options(Wp,-D_GLIBCXX_ASSERTIONS)
+add_compile_options(-w)
 # apt-ftparchive dependencies
 find_package(Berkeley REQUIRED)
 if (BERKELEY_FOUND)
@@ -124,21 +121,6 @@
 endif()
 
 
-find_package(Udev)
-if (UDEV_FOUND)
-  set(HAVE_UDEV 1)
-endif()
-
-find_package(Systemd)
-if (SYSTEMD_FOUND)
-  set(HAVE_SYSTEMD 1)
-endif()
-
-find_package(SECCOMP)
-if (SECCOMP_FOUND)
-  set(HAVE_SECCOMP 1)
-endif()
-
 find_package(XXHASH REQUIRED)
 
 # Mount()ing and stat()ing and friends
@@ -255,7 +237,6 @@
 add_subdirectory(dselect)
 add_subdirectory(ftparchive)
 add_subdirectory(methods)
-add_subdirectory(test)
 
 if (USE_NLS)
 add_subdirectory(po)
diff -urN a/ftparchive/byhash.cc b/ftparchive/byhash.cc
--- a/ftparchive/byhash.cc
+++ b/ftparchive/byhash.cc
@@ -15,6 +15,8 @@
 #include <string>
 
 #include <sys/stat.h>
+
+#define st_mtim st_mtimespec
 #include <unistd.h>
 
 #include "byhash.h"
diff -urN a/methods/connect.cc b/methods/connect.cc
--- a/methods/connect.cc
+++ b/methods/connect.cc
@@ -277,7 +277,7 @@
    struct timeval tv = {
       // Split our millisecond timeout into seconds and microseconds
       .tv_sec = TimeoutMsec / 1000,
-      .tv_usec = (TimeoutMsec % 1000) * 1000,
+      .tv_usec = static_cast<__darwin_suseconds_t>((TimeoutMsec % 1000) * 1000),
    };
 
    // We will return once we have no more connections, a time out, or
