--- src/data/tupletable.orig
+++ src/data/tupletable
@@ -55,8 +55,8 @@
 base-bsd-openbsd-<cpu>		openbsd-<cpu>
 base-bsd-netbsd-<cpu>		netbsd-<cpu>
 base-bsd-darwin-amd64		darwin-amd64
-base-bsd-darwin-arm		darwin-arm
-base-bsd-darwin-arm64		darwin-arm64
+base-bsd-darwin-arm		iphoneos-arm
+base-bsd-darwin-arm64		@DPKGARCH@
 base-bsd-darwin-i386		darwin-i386
 base-bsd-darwin-powerpc		darwin-powerpc
 base-bsd-darwin-ppc64		darwin-ppc64
