--- src/src/unix/core.c.orig
+++ src/src/unix/core.c
@@ -1893,7 +1893,7 @@
     rc = 1;
 
   return (unsigned) rc;
-#else  /* __linux__ */
+#elif defined(_SC_NPROCESSORS_ONLN)
   long rc;
 
   rc = sysconf(_SC_NPROCESSORS_ONLN);
@@ -1901,5 +1901,17 @@
     rc = 1;
 
   return (unsigned) rc;
+#elif defined(CTL_HW) && defined(HW_NCPU)
+  int mib[2], ncpus = 0;
+  mib[0]     = CTL_HW;
+  mib[1]     = HW_NCPU;
+  size_t len = sizeof(ncpus);
+  sysctl(mib, 2, &ncpus, &len, NULL, 0);
+  if (ncpus < 1)
+    return 1;
+  return (unsigned) ncpus;
+#else
+#warning No known way to determine number of CPUs
+  return 1;
 #endif  /* __linux__ */
 }
