--- src/log.c.orig
+++ src/log.c
@@ -54,7 +54,8 @@
 	error = mkdir(dataPath(buf, sizeof(buf), "log", 0), S_IRWXU);
 	if (error && errno != EEXIST) err(EX_CANTCREAT, "%s", buf);
 
-	logDir = open(buf, O_RDONLY | O_CLOEXEC);
+	logDir = open(buf, O_RDONLY);
+	fcntl(logDir, F_SETFD, FD_CLOEXEC);
 	if (logDir < 0) err(EX_CANTCREAT, "%s", buf);
 
 #ifdef __FreeBSD__
@@ -123,9 +124,10 @@
 
 	int fd = openat(
 		logDir, path,
-		O_WRONLY | O_APPEND | O_CREAT | O_CLOEXEC,
+		O_WRONLY | O_APPEND | O_CREAT,
 		S_IRUSR | S_IWUSR
 	);
+	fcntl(fd, F_SETFD, FD_CLOEXEC);
 	if (fd < 0) err(EX_CANTCREAT, "log/%s", path);
 	logs[id].file = fdopen(fd, "a");
 	if (!logs[id].file) err(EX_OSERR, "fdopen");
