--- src/log.c.orig
+++ src/log.c
@@ -54,7 +54,12 @@
 	error = mkdir(dataPath(buf, sizeof(buf), "log", 0), S_IRWXU);
 	if (error && errno != EEXIST) err(EX_CANTCREAT, "%s", buf);
 
+#ifdef O_CLOEXEC
 	logDir = open(buf, O_RDONLY | O_CLOEXEC);
+#else
+	logDir = open(buf, O_RDONLY);
+	fcntl(logDir, F_SETFD, FD_CLOEXEC);
+#endif
 	if (logDir < 0) err(EX_CANTCREAT, "%s", buf);
 
 #ifdef __FreeBSD__
@@ -123,9 +128,16 @@
 
 	int fd = openat(
 		logDir, path,
+#ifdef O_CLOEXEC
 		O_WRONLY | O_APPEND | O_CREAT | O_CLOEXEC,
+#else
+		O_WRONLY | O_APPEND | O_CREAT,
+#endif
 		S_IRUSR | S_IWUSR
 	);
+#ifndef O_CLOEXEC
+	fcntl(fd, F_SETFD, FD_CLOEXEC);
+#endif
 	if (fd < 0) err(EX_CANTCREAT, "log/%s", path);
 	logs[id].file = fdopen(fd, "a");
 	if (!logs[id].file) err(EX_OSERR, "fdopen");
