--- src/ex.c.orig
+++ src/ex.c
@@ -15,7 +15,7 @@
 int xorder = 1;			/* change the order of characters */
 int xkmap;			/* the current keymap */
 int xkmap_alt = 1;		/* the alternate keymap */
-int xtabspc = 8;		/* number of spaces for tab */
+int xtabspc = 4;		/* number of spaces for tab */
 int xish = 1;			/* interactive shell */
 int xgrp = 2;			/* regex search group */
 int xpac;			/* print autocomplete options */
