--- src/neofetch.orig
+++ src/neofetch
@@ -3674,6 +3674,8 @@
                 pages_compressed="${pages_compressed:-0}"
                 mem_used="$(((pages_app + ${pages_wired//.} + ${pages_compressed//.}) * hw_pagesize / 1024))"
             fi
+            mem_total="$(($(sysctl -n hw.memsize) / 1024))"
+            type -p getusedmem &>/dev/null && mem_used="$(getusedmem)"
         ;;
 
         "BSD" | "MINIX" | "ravynOS")
