--- src/neofetch.orig
+++ src/neofetch
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/var/usr/bin/env bash
 # vim: noai:ts=4:sw=4:expandtab
 # shellcheck source=/dev/null
 # shellcheck disable=2009 disable=2153 disable=2154 disable=2243 disable=2244 disable=2317
@@ -193,7 +193,7 @@
 # kib  '1020928KiB / 7117824KiB'
 # mib  '1042MiB / 6951MiB'
 # gib: ' 0.98GiB / 6.79GiB'
-memory_unit="gib"
+memory_unit="mib"
 
 # Change memory output precision.
 #
@@ -3423,13 +3423,17 @@
         ;;
 
         "Mac OS X" | "macOS" | "iPhone OS")
-            hw_pagesize="$(sysctl -n hw.pagesize)"
             mem_total="$(($(sysctl -n hw.memsize) / 1024))"
+            if type -p getusedmem > /dev/null; then
+                mem_used="$(getusedmem)"
+            else
+            hw_pagesize="$(sysctl -n hw.pagesize)"
             pages_app="$(($(sysctl -n vm.page_pageable_internal_count) - $(sysctl -n vm.page_purgeable_count)))"
             pages_wired="$(vm_stat | awk '/ wired/ { print $4 }')"
             pages_compressed="$(vm_stat | awk '/ occupied/ { printf $5 }')"
             pages_compressed="${pages_compressed:-0}"
             mem_used="$(((pages_app + ${pages_wired//.} + ${pages_compressed//.}) * hw_pagesize / 1024))"
+            fi
         ;;
 
         "BSD" | "MINIX" | "ravynOS")
