--- source/neofetch.orig
+++ source/neofetch
@@ -3266,6 +3266,10 @@
         ;;
 
         "Mac OS X" | "macOS" | "iPhone OS")
+            if command -v getusedmem >/dev/null; then
+                mem_used="$(getusedmem)"
+                mem_total="$(sysctl -n hw.memsize)"
+            else
             hw_pagesize="$(sysctl -n hw.pagesize)"
             mem_total="$(($(sysctl -n hw.memsize) / 1024))"
             pages_app="$(($(sysctl -n vm.page_pageable_internal_count) - $(sysctl -n vm.page_purgeable_count)))"
@@ -3273,6 +3277,7 @@
             pages_compressed="$(vm_stat | awk '/ occupied/ { printf $5 }')"
             pages_compressed="${pages_compressed:-0}"
             mem_used="$(((pages_app + ${pages_wired//.} + ${pages_compressed//.}) * hw_pagesize / 1024))"
+            fi
         ;;
 
         "BSD" | "MINIX" | "ravynOS")
