--- src/Modules/posixmodule.c.orig
+++ src/Modules/posixmodule.c
@@ -9,6 +9,9 @@
 
 // --- Python includes ------------------------------------------------------
 
+#ifdef __APPLE__
+#include <sys/socket.h>
+#endif
 #include "Python.h"
 
 #ifdef __VXWORKS__
@@ -451,11 +454,6 @@
 
 #include <mach/mach.h>
 
-#if defined(__has_builtin)
-#if __has_builtin(__builtin_available)
-#define HAVE_BUILTIN_AVAILABLE 1
-#endif
-#endif
 
 #ifdef HAVE_BUILTIN_AVAILABLE
 #  define HAVE_FSTATAT_RUNTIME __builtin_available(macOS 10.10, iOS 8.0, *)
