--- src/Modules/posixmodule.c.orig
+++ src/Modules/posixmodule.c
@@ -7,6 +7,9 @@
    of the compiler used.  Different compilers define their own feature
    test macro, e.g. '_MSC_VER'. */
 
+#ifdef __APPLE__
+#include <sys/socket.h>
+#endif
 #include "Python.h"
 
 #ifdef __VXWORKS__
@@ -101,11 +104,6 @@
 
 #include <mach/mach.h>
 
-#if defined(__has_builtin)
-#if __has_builtin(__builtin_available)
-#define HAVE_BUILTIN_AVAILABLE 1
-#endif
-#endif
 
 #ifdef HAVE_BUILTIN_AVAILABLE
 #  define HAVE_FSTATAT_RUNTIME __builtin_available(macOS 10.10, iOS 8.0, *)
