--- src/configure.ac.orig
+++ src/configure.ac
@@ -737,53 +737,6 @@
 AC_SUBST([APP_STORE_COMPLIANCE_PATCH])
 
 AC_SUBST([_PYTHON_HOST_PLATFORM])
-if test "$cross_compiling" = yes; then
-	case "$host" in
-	*-*-linux*)
-		case "$host_cpu" in
-		arm*)
-			_host_ident=arm
-			;;
-		*)
-			_host_ident=$host_cpu
-		esac
-		;;
-	*-*-cygwin*)
-		_host_ident=
-		;;
-	*-apple-ios*)
-		_host_os=`echo $host | cut -d '-' -f3`
-		_host_device=`echo $host | cut -d '-' -f4`
-		_host_device=${_host_device:=os}
-
-		# IPHONEOS_DEPLOYMENT_TARGET is the minimum supported iOS version
-		AC_MSG_CHECKING([iOS deployment target])
-		IPHONEOS_DEPLOYMENT_TARGET=$(echo ${_host_os} | cut -c4-)
-		IPHONEOS_DEPLOYMENT_TARGET=${IPHONEOS_DEPLOYMENT_TARGET:=13.0}
-		AC_MSG_RESULT([$IPHONEOS_DEPLOYMENT_TARGET])
-
-		case "$host_cpu" in
-			aarch64)
-				_host_ident=${IPHONEOS_DEPLOYMENT_TARGET}-arm64-iphone${_host_device}
-				;;
-			*)
-				_host_ident=${IPHONEOS_DEPLOYMENT_TARGET}-$host_cpu-iphone${_host_device}
-				;;
-		esac
-		;;
-	*-*-vxworks*)
-		_host_ident=$host_cpu
-		;;
-	wasm32-*-* | wasm64-*-*)
-		_host_ident=$host_cpu
-		;;
-	*)
-		# for now, limit cross builds to known configurations
-		MACHDEP="unknown"
-		AC_MSG_ERROR([cross build not supported for $host])
-	esac
-	_PYTHON_HOST_PLATFORM="$MACHDEP${_host_ident:+-$_host_ident}"
-fi
 
 # Some systems cannot stand _XOPEN_SOURCE being defined at all; they
 # disable features if it is defined, without any means to access these
