--- src/configure.ac.orig
+++ src/configure.ac
@@ -4077,19 +4077,6 @@
 
 dnl detect libffi
 have_libffi=missing
-AS_VAR_IF([ac_sys_system], [Darwin], [
-  WITH_SAVE_ENV([
-    CFLAGS="-I${SDKROOT}/usr/include/ffi $CFLAGS"
-    AC_CHECK_HEADER([ffi.h], [
-      AC_CHECK_LIB([ffi], [ffi_call], [
-        dnl use ffi from SDK root
-        have_libffi=yes
-        LIBFFI_CFLAGS="-I${SDKROOT}/usr/include/ffi -DUSING_APPLE_OS_LIBFFI=1"
-        LIBFFI_LIBS="-lffi"
-      ])
-    ])
-  ])
-])
 AS_VAR_IF([have_libffi], [missing], [
   PKG_CHECK_MODULES([LIBFFI], [libffi], [have_libffi=yes], [
     WITH_SAVE_ENV([
@@ -4109,13 +4096,6 @@
 AS_VAR_IF([have_libffi], [yes], [
   ctypes_malloc_closure=no
   AS_CASE([$ac_sys_system],
-    [Darwin], [
-      dnl when do we need USING_APPLE_OS_LIBFFI?
-      ctypes_malloc_closure=yes
-    ],
-    [iOS], [
-      ctypes_malloc_closure=yes
-    ],
     [sunos5], [AS_VAR_APPEND([LIBFFI_LIBS], [" -mimpure-text"])]
   )
   AS_VAR_IF([ctypes_malloc_closure], [yes], [
