--- src/Include/pythread.h.orig
+++ src/Include/pythread.h
@@ -20,7 +20,7 @@
 PyAPI_FUNC(void) _Py_NO_RETURN PyThread_exit_thread(void);
 PyAPI_FUNC(unsigned long) PyThread_get_thread_ident(void);
 
-#if (defined(__APPLE__) || defined(__linux__) || defined(_WIN32) \
+#if (defined(__linux__) || defined(_WIN32) \
      || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) \
      || defined(__OpenBSD__) || defined(__NetBSD__) \
      || defined(__DragonFly__) || defined(_AIX))
