--- src/Lib/posixpath.py.orig
+++ src/Lib/posixpath.py
@@ -18,7 +18,7 @@
 extsep = '.'
 sep = '/'
 pathsep = ':'
-defpath = '/bin:/usr/bin'
+defpath = '/var/usr/bin:/bin:/usr/bin'
 altsep = None
 devnull = '/dev/null'
 
--- src/Lib/uuid.py.orig
+++ src/Lib/uuid.py
@@ -426,7 +426,7 @@
 
     try:
         path_dirs = os.environ.get('PATH', os.defpath).split(os.pathsep)
-        path_dirs.extend(['/sbin', '/usr/sbin'])
+        path_dirs.extend(['/var/usr/sbin', '/sbin', '/usr/sbin'])
         executable = shutil.which(command, path=os.pathsep.join(path_dirs))
         if executable is None:
             return None
--- src/Lib/subprocess.py.orig
+++ src/Lib/subprocess.py
@@ -1846,7 +1846,7 @@
             if shell:
                 # On Android the default shell is at '/system/bin/sh'.
                 unix_shell = ('/system/bin/sh' if
-                          hasattr(sys, 'getandroidapilevel') else '/bin/sh')
+                          hasattr(sys, 'getandroidapilevel') else '/var/usr/bin/sh')
                 args = [unix_shell, "-c"] + args
                 if executable:
                     args[0] = executable
