--- src/Python/pylifecycle.c.orig
+++ src/Python/pylifecycle.c
@@ -49,7 +49,11 @@
 // The os_log unified logging APIs were introduced in macOS 10.12, iOS 10.0,
 // tvOS 10.0, and watchOS 3.0;
 #  if defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE
-#    define HAS_APPLE_SYSTEM_LOG 1
+#    if defined(__IPHONE_10_0) && __IPHONE_OS_VERSION_MIN_REQUIRED >= __IPHONE_10_0
+#      define HAS_APPLE_SYSTEM_LOG 1
+#    else
+#      define HAS_APPLE_SYSTEM_LOG 0
+#    endif
 #  elif defined(TARGET_OS_OSX) && TARGET_OS_OSX
 #    if defined(MAC_OS_X_VERSION_10_12) && MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_12
 #      define HAS_APPLE_SYSTEM_LOG 1
