--- src/Python/thread_pthread.h.orig
+++ src/Python/thread_pthread.h
@@ -387,8 +387,15 @@
     if (!initialized)
         PyThread_init_thread();
 #ifdef __APPLE__
+#if (defined(__ENVIRONMENT_IPHONE_OS_VERSION_MIN_REQUIRED__) &&                \
+     __ENVIRONMENT_IPHONE_OS_VERSION_MIN_REQUIRED__ < 30200) ||                \
+    (defined(__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__) &&                 \
+     __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 1060)
+    mach_port_t native_id = pthread_mach_thread_np(pthread_self());
+#else
     uint64_t native_id;
     (void) pthread_threadid_np(NULL, &native_id);
+#endif
 #elif defined(__linux__)
     pid_t native_id;
     native_id = syscall(SYS_gettid);
