--- src/src/lj_prng.c.orig
+++ src/src/lj_prng.c
@@ -121,8 +121,7 @@
 ** off the foolery, /dev/urandom is handled by the same kernel code,
 ** yet accessing it is actually permitted (but less efficient).
 */
-#include <Availability.h>
-#if __MAC_OS_X_VERSION_MIN_REQUIRED >= 101200
+#if __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 101200
 #define LJ_TARGET_HAS_GETENTROPY	1
 #endif
 #elif (LJ_TARGET_BSD && !defined(__NetBSD__)) || LJ_TARGET_SOLARIS || LJ_TARGET_CYGWIN || LJ_TARGET_QNX
