--- src/src/socket.c.orig
+++ src/src/socket.c
@@ -41,9 +41,6 @@
 #ifndef IFF_RUNNING
 #define IFF_RUNNING IFF_UP
 #endif
-#ifndef AI_NUMERICSERV
-#define AI_NUMERICSERV 0
-#endif
 #else
 #include <sys/socket.h>
 #include <sys/un.h>
@@ -68,6 +65,9 @@
 #ifdef HAVE_POLL
 #include <sys/poll.h>
 #endif
+#ifndef AI_NUMERICSERV
+#define AI_NUMERICSERV 0
+#endif
 
 #define RECV_TIMEOUT 20000
 #define SEND_TIMEOUT 10000
