--- src/llvm/include/llvm/TargetParser/Triple.h.orig
+++ src/llvm/include/llvm/TargetParser/Triple.h
@@ -1080,7 +1080,15 @@
   /// Note: Android API level 29 (10) introduced ELF TLS.
   bool hasDefaultEmulatedTLS() const {
     return (isAndroid() && isAndroidVersionLT(29)) || isOSOpenBSD() ||
-           isWindowsCygwinEnvironment() || isOHOSFamily();
+           isWindowsCygwinEnvironment() || isOHOSFamily() ||
+           (isMacOSX() && isMacOSXVersionLT(10, 7)) ||
+           (isiOS() && ((isArch64Bit() && isOSVersionLT(8)) ||
+                        (isArch32Bit() &&
+                         ((isSimulatorEnvironment() && isOSVersionLT(10)) ||
+                          isOSVersionLT(9))))) ||
+           (isWatchOS() && (isOSVersionLT(2) ||
+                            (isSimulatorEnvironment() && isOSVersionLT(3)))) ||
+           isDriverKit();
   }
 
   /// True if the target supports both general-dynamic and TLSDESC, and TLSDESC

