--- src/llvm/lib/Support/CrashRecoveryContext.cpp.orig
+++ src/llvm/lib/Support/CrashRecoveryContext.cpp
@@ -478,13 +478,13 @@
 
 // FIXME: Portability.
 static void setThreadBackgroundPriority() {
-#ifdef __APPLE__
+#ifdef PRIO_DARWIN_THREAD
   setpriority(PRIO_DARWIN_THREAD, 0, PRIO_DARWIN_BG);
 #endif
 }
 
 static bool hasThreadBackgroundPriority() {
-#ifdef __APPLE__
+#ifdef PRIO_DARWIN_THREAD
   return getpriority(PRIO_DARWIN_THREAD, 0) == 1;
 #else
   return false;
