--- src/core/shared/platform/common/posix/posix_thread.c.orig
+++ src/core/shared/platform/common/posix/posix_thread.c
@@ -13,6 +13,7 @@
 
 #if defined(__APPLE__) || defined(__MACH__)
 #include <TargetConditionals.h>
+#include <iphoneports/sigreturn.h>
 #endif
 
 typedef struct {
@@ -773,7 +774,6 @@
 #if WASM_DISABLE_STACK_HW_BOUND_CHECK == 0
 #if defined(__APPLE__)
 #define UC_RESET_ALT_STACK 0x80000000
-    extern int __sigreturn(void *, int);
 
     /* It's necessary to call __sigreturn to restore the sigaltstack state
        after exiting the signal handler. */
