--- src/gnu/stdio.in.h.orig
+++ src/gnu/stdio.in.h
@@ -37,7 +37,7 @@
 #ifndef _@GUARD_PREFIX@_STDIO_H
 
 /* Suppress macOS deprecation warnings for sprintf and vsprintf.  */
-#if (defined __APPLE__ && defined __MACH__) && !defined _POSIX_C_SOURCE
+#if (defined __APPLE__ && defined __MACH__) && !defined _POSIX_C_SOURCE && !(defined(__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__) && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 1050)
 # define _POSIX_C_SOURCE 200809L
 # define _GL_DEFINED__POSIX_C_SOURCE
 #endif
