--- src/MagickCore/version.c.orig
+++ src/MagickCore/version.c
@@ -663,12 +663,12 @@
 #if defined(MAGICKCORE_MSC_VER)
   (void) FormatLocaleFile(file,"Compiler: Visual Studio %d (%d)\n",
     MAGICKCORE_MSC_VER,_MSC_FULL_VER);
-#elif defined(__GNUC__)
-  (void) FormatLocaleFile(file,"Compiler: gcc (%d.%d)\n",__GNUC__,
-    __GNUC_MINOR__);
 #elif defined(__clang__)
   (void) FormatLocaleFile(file,"Compiler: clang (%d.%d.%d)\n",__clang_major__,
     __clang_minor__,__clang_patchlevel__);
+#elif defined(__GNUC__)
+  (void) FormatLocaleFile(file,"Compiler: gcc (%d.%d)\n",__GNUC__,
+    __GNUC_MINOR__);
 #elif defined(__MINGW32_MAJOR_VERSION)
   (void) FormatLocaleFile(file,"Compiler: MinGW (%d.%d)\n",
     __MINGW32_MAJOR_VERSION,__MINGW32_MINOR_VERSION);
