--- src/thread_pthread.c.orig
+++ src/thread_pthread.c
@@ -2759,9 +2759,7 @@
     is not defined on 10.5, and while __POWERPC__ takes care of ppc/ppc64,
     i386 will be broken without this. Note, 10.5 is supported with GCC upstream,
     so it has C++17 and everything needed to build modern Ruby. */
-# if (!defined(MAC_OS_X_VERSION_10_6) || \
-      (MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_6) || \
-      defined(__POWERPC__) /* never defined for PowerPC platforms */)
+# if (1)
     const bool no_pthread_threadid_np = true;
 #   define NO_PTHREAD_MACH_THREAD_NP 1
 # elif MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_6
