--- src/os.hpp.orig
+++ src/os.hpp
@@ -222,7 +222,13 @@
 #endif
 
 // Unlike Apple x64, utimensat shall be available in all Apple M1 systems.
-#if _POSIX_C_SOURCE >= 200809L || defined(__APPLE__) && defined(__arm64__)
+#if ((defined(__ENVIRONMENT_IPHONE_OS_VERSION_MIN_REQUIRED__) || \
+      defined(__ENVIRONMENT_TV_OS_VERSION_MIN_REQUIRED__)) &&    \
+     __ENVIRONMENT_OS_VERSION_MIN_REQUIRED__ >= 110000) ||       \
+    (defined(__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__) &&   \
+     __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 101300) || \
+    (defined(__ENVIRONMENT_WATCH_OS_VERSION_MIN_REQUIRED__) &&   \
+     __ENVIRONMENT_WATCH_OS_VERSION_MIN_REQUIRED__ >= 40000)
   #define UNIX_TIME_NS // Nanosecond time precision in Unix.
 #endif
 
