--- src/gzip.c.orig
+++ src/gzip.c
@@ -78,6 +78,31 @@
 #include <sys/attr.h>
 #include <copyfile.h>
 #include <get_compat.h>
+
+char *strndup(const char *str, size_t maxlen) {
+	const char *end = memchr(str, '\0', maxlen);
+	size_t len = end ? end - str : maxlen;
+	char *newstr = malloc(len + 1);
+	if (!newstr) return NULL;
+	memcpy(newstr, str, len);
+	newstr[len] = '\0';
+	return newstr;
+}
+
+int fgetattrlist(int fd, struct attrlist *attrList, void *attrBuf, size_t attrBufSize, unsigned long options) {
+	char fdpath[PATH_MAX];
+	if (fcntl(fd, F_GETPATH, fdpath) == -1)
+		return -1;
+	return getattrlist(fdpath, attrList, attrBuf, attrBufSize, options);
+}
+
+int fsetattrlist(int fd, struct attrlist *attrList, void *attrBuf, size_t attrBufSize, unsigned long options) {
+	char fdpath[PATH_MAX];
+	if (fcntl(fd, F_GETPATH, fdpath) == -1)
+		return -1;
+	return setattrlist(fdpath, attrList, attrBuf, attrBufSize, options);
+}
+
 int futimens(int fd, const struct timespec times[2]);
 
 #define	st_atim	st_atimespec
