--- src/src/ccache/util/file.cpp.orig
+++ src/src/ccache/util/file.cpp
@@ -32,7 +32,7 @@
 #include <ccache/util/pathstring.hpp>
 #include <ccache/util/temporaryfile.hpp>
 
-#ifdef __APPLE__
+#if defined(__APPLE__) && __has_include(<copyfile.h>)
 #  include <copyfile.h>
 #endif
 
@@ -152,7 +152,7 @@
     }
   }
 
-#  if defined(__APPLE__)
+#  if defined(__APPLE__) && __has_include(<copyfile.h>)
   copyfile_state_t state = copyfile_state_alloc();
   int n = fcopyfile(*src_fd, *dst_fd, state, COPYFILE_DATA);
   copyfile_state_free(state);
