--- src/src/ccache/execute.cpp.orig
+++ src/src/ccache/execute.cpp
@@ -287,6 +287,31 @@
 {
   LOG("Executing {}", util::format_argv_for_logging(argv));
 
+#if (defined(__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__) && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 1050) || \
+    (defined(__ENVIRONMENT_IPHONE_OS_VERSION_MIN_REQUIRED__) && __ENVIRONMENT_IPHONE_OS_VERSION_MIN_REQUIRED__ < 20000)
+  {
+    SignalHandlerBlocker signal_handler_blocker;
+    ctx.compiler_pid = fork();
+  }
+
+  if (ctx.compiler_pid == -1) {
+    throw core::Fatal(FMT("Failed to fork: {}", strerror(errno)));
+  }
+
+  if (ctx.compiler_pid == 0) {
+    // Child.
+    dup2(*fd_out, STDOUT_FILENO);
+    fd_out.close();
+    dup2(*fd_err, STDERR_FILENO);
+    fd_err.close();
+    exit(execv(argv[0], const_cast<char* const*>(argv)));
+  }
+
+  fd_out.close();
+  fd_err.close();
+
+#else
+
   util::Fd out(std::move(fd_out));
   util::Fd err(std::move(fd_err));
 
@@ -335,6 +360,8 @@
     return -1;
   }
 
+#endif
+
   int status;
   while (waitpid(ctx.compiler_pid, &status, 0) == -1) {
     if (errno != EINTR) {
--- src/src/ccache/util/exec.cpp.orig
+++ src/src/ccache/util/exec.cpp
@@ -118,6 +118,30 @@
     return tl::unexpected(FMT("Non-zero exit code: {}", exitcode));
   }
 #else
+#if (defined(__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__) && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 1050) || \
+    (defined(__ENVIRONMENT_IPHONE_OS_VERSION_MIN_REQUIRED__) && __ENVIRONMENT_IPHONE_OS_VERSION_MIN_REQUIRED__ < 20000)
+  int pipefd[2];
+  if (pipe(pipefd) == -1) {
+    return tl::unexpected(FMT("pipe failed: {}", strerror(errno)));
+  }
+
+  pid_t pid = fork();
+  if (pid == -1) {
+    return tl::unexpected(FMT("fork failed: {}", strerror(errno)));
+  }
+
+  if (pid == 0) {
+    // Child.
+    close(pipefd[0]);
+    close(0);
+    dup2(pipefd[1], 1);
+    dup2(pipefd[1], 2);
+    _exit(execvp(argv[0], const_cast<char* const*>(argv.data())));
+    // Never reached.
+  }
+  // Parent
+  close(pipefd[1]);
+#else
   int pipefd[2];
   CHECK_LIB_CALL(pipe, pipefd);
 
@@ -141,6 +165,7 @@
     return tl::unexpected(
       FMT("posix_spawnp failed: {}", strerror(saved_errno)));
   }
+#endif
 
   auto read_result = read_fd(pipefd[0], [&](auto data) {
     output.append(reinterpret_cast<const char*>(data.data()), data.size());
