--- src/src/ccache/util/file.cpp.orig
+++ src/src/ccache/util/file.cpp
@@ -54,7 +54,7 @@
 #endif
 
 #ifdef HAVE_UTIMENSAT
-#elif defined(HAVE_UTIMES)
+#elif 0
 #  include <sys/time.h>
 #else
 #  ifdef HAVE_UTIME_H
@@ -551,7 +551,7 @@
   }
   utimensat(
     AT_FDCWD, util::pstr(path).c_str(), mtime ? atime_mtime : nullptr, 0);
-#elif defined(HAVE_UTIMES)
+#elif 0
   timeval atime_mtime[2];
   if (mtime) {
     atime_mtime[0].tv_sec = atime ? util::sec(*atime) : util::sec(*mtime);
