--- src/cctools/ld64/src/abstraction/MachOFileAbstraction.hpp.orig
+++ src/cctools/ld64/src/abstraction/MachOFileAbstraction.hpp
@@ -572,7 +572,7 @@
 	#define SUPPORT_ARCH_arm_any 1
 #endif
 #if SUPPORT_ARCH_armv6
-	{ "armv6", CPU_TYPE_ARM,     CPU_SUBTYPE_ARM_V6,     "armv6-",   "", true,  Thumb2Support::none },
+	{ "armv6", CPU_TYPE_ARM,     CPU_SUBTYPE_ARM_V6,     "armv6-",   "armv6k-", true,  Thumb2Support::none },
 	#define SUPPORT_ARCH_arm_any 1
 #endif
 #if SUPPORT_ARCH_armv7
--- src/cctools/libstuff/lto.c.orig
+++ src/cctools/libstuff/lto.c
@@ -251,6 +251,7 @@
 	    arch_flag->cpusubtype = CPU_SUBTYPE_ARM_V5TEJ;
 	}
 	else if(strncmp(target_triple, "armv6", n) == 0 ||
+	        strncmp(target_triple, "armv6k", n) == 0 ||
 	        strncmp(target_triple, "thumbv6", n) == 0){
 	    arch_flag->cputype = CPU_TYPE_ARM;
 	    arch_flag->cpusubtype = CPU_SUBTYPE_ARM_V6;
