--- src/main.c.orig
+++ src/main.c
@@ -35,7 +35,6 @@
 #endif /* not lint */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
 
 /*
  * CREDITS
--- src/io.c.orig
+++ src/io.c
@@ -26,7 +26,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
 
 #include "ed.h"
 
--- src/buf.c.orig
+++ src/buf.c
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
 
 #include <sys/file.h>
 #include <sys/stat.h>
--- src/re.c.orig
+++ src/re.c
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
 
 #include "ed.h"
 
--- src/glbl.c.orig
+++ src/glbl.c
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
 
 #include <sys/types.h>
 #include <sys/ioctl.h>
--- src/undo.c.orig
+++ src/undo.c
@@ -26,7 +26,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
 
 #include "ed.h"
 
--- src/sub.c.orig
+++ src/sub.c
@@ -27,7 +27,6 @@
  */
 
 #include <sys/cdefs.h>
-__FBSDID("$FreeBSD$");
 
 #include "ed.h"
 
