--- src/ar.sh.orig
+++ src/ar.sh
@@ -47,7 +47,7 @@
 	    local archdir="${dir}/${arch}";
 	    mkdir -p "${archdir}";
 
-	    lipo -thin "${arch}" "${archive}" -o "${archdir}/${name}";
+	    "$_TARGET-lipo" -thin "${arch}" "${archive}" -o "${archdir}/${name}";
 
 	    ( cd "${archdir}" && ar -xo "./${name}"; );
 
@@ -62,7 +62,7 @@
 
 	local ofile;
 	for ofile in ${ofiles}; do
-	    lipo -create $(find "${dir}" -name "${ofile}" -print) -o "${dir}/${ofile}";
+	    "$_TARGET-lipo" -create $(find "${dir}" -name "${ofile}" -print) -o "${dir}/${ofile}";
 	done
 
 	( cd "${dir}" && ar -cr "${name}" ${ofiles}; );
