--- src/arch/arm/neon_intrins.h.orig
+++ src/arch/arm/neon_intrins.h
@@ -25,7 +25,7 @@
     out.val[3] = vqsubq_u16(a.val[3], b); \
 } while (0)
 
-#  if defined(__clang__) && defined(__arm__) && defined(__ANDROID__)
+#  if defined(__clang__) && defined(__arm__) && (defined(__ANDROID__) || defined(__APPLE__))
 /* Clang for 32-bit Android has too strict alignment requirement (:256) for x4 NEON intrinsics */
 #    undef ARM_NEON_HASLD4
 #    undef vld1q_u16_x4
