--- src/packaging/RedHat/conf/commands.allow.sh.orig
+++ src/packaging/RedHat/conf/commands.allow.sh
@@ -38,6 +38,16 @@
   /usr/bin/g++
   /usr/bin/*gcc-*
   /usr/bin/*g++-*
+  /usr/bin/arm-apple-darwin9-gcc
+  /usr/bin/arm-apple-darwin9-g++
+  /usr/bin/clang
+  /usr/bin/clang++
+  /var/usr/bin/clang
+  /var/usr/bin/clang++
+  /var/usr/bin/cc
+  /var/usr/bin/c++
+  /var/usr/bin/gcc
+  /var/usr/bin/g++
 "
 
 # You shouldn't need to alter anything below here.
