--- src/llvm/include/llvm/Support/Compiler.h.orig
+++ src/llvm/include/llvm/Support/Compiler.h
@@ -534,7 +534,7 @@
 /// initialize to some constant value. In almost all circumstances this is most
 /// appropriate for use with a pointer, integer, or small aggregation of
 /// pointers and integers.
-#if LLVM_ENABLE_THREADS
+#if 0
 #if __has_feature(cxx_thread_local) || defined(_MSC_VER)
 #define LLVM_THREAD_LOCAL thread_local
 #else
