--- src/src/llvm/lib/Support/CrashRecoveryContext.cpp.orig
+++ src/src/llvm/lib/Support/CrashRecoveryContext.cpp
@@ -465,13 +465,13 @@
 
 // FIXME: Portability.
 static void setThreadBackgroundPriority() {
-#ifdef __APPLE__
+#ifdef PRIO_DARWIN_THREAD
   setpriority(PRIO_DARWIN_THREAD, 0, PRIO_DARWIN_BG);
 #endif
 }
 
 static bool hasThreadBackgroundPriority() {
-#ifdef __APPLE__
+#ifdef PRIO_DARWIN_THREAD
   return getpriority(PRIO_DARWIN_THREAD, 0) == 1;
 #else
   return false;
--- src/src/llvm/lib/Support/Unix/Threading.inc.orig
+++ src/src/llvm/lib/Support/Unix/Threading.inc
@@ -259,7 +259,7 @@
              &priority)
              ? SetThreadPriorityResult::SUCCESS
              : SetThreadPriorityResult::FAILURE;
-#elif defined(__APPLE__)
+#elif defined(__APPLE__) && defined(PRIO_DARWIN_THREAD)
   // https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/getpriority.2.html
   // When setting a thread into background state the scheduling priority is set
   // to lowest value, disk and network IO are throttled. Network IO will be
