--- src/src/mkheader.c.orig
+++ src/src/mkheader.c
@@ -618,7 +618,7 @@
       if (cross_building
           || try_include_file (fname, lnr,
                                "./lock-obj-pub.native.h", write_line))
-        include_file (fname, lnr, "syscfg/lock-obj-pub.&.h", write_line);
+        include_file (fname, lnr, "syscfg/lock-obj-pub.cross-os.h", write_line);
     }
   else
     return 0; /* Unknown tag.  */
