--- src/darwin/PlatformHelpers.c.orig
+++ src/darwin/PlatformHelpers.c
@@ -60,6 +60,8 @@
 
 void Platform_getCPUBrandString(char* cpuBrandString, size_t cpuBrandStringSize) {
    if (sysctlbyname("machdep.cpu.brand_string", cpuBrandString, &cpuBrandStringSize, NULL, 0) == -1) {
+      if (sysctlbyname("hw.cpusubtype", cpuBrandString, &cpuBrandStringSize, NULL, 0) != -1)
+         return;
       fprintf(stderr,
          "WARN: Unable to determine the CPU brand string.\n"
          "errno: %i, %s\n", errno, strerror(errno));
