--- src/bzmore.orig
+++ src/bzmore
@@ -1,9 +1,9 @@
-#!/bin/sh
+#!/var/usr/bin/sh
 
 # Bzmore wrapped for bzip2, 
 # adapted from zmore by Philippe Troin <phil@fifi.org> for Debian GNU/Linux.
 
-PATH="/usr/bin:$PATH"; export PATH
+PATH="/var/usr/bin:/usr/bin:$PATH"; export PATH
 
 prog=`echo $0 | sed 's|.*/||'`
 case "$prog" in
--- src/bzdiff.orig
+++ src/bzdiff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/var/usr/bin/sh
 # sh is buggy on RS/6000 AIX 3.2. Replace above line with #!/bin/ksh
 
 # Bzcmp/diff wrapped for bzip2, 
@@ -12,7 +12,7 @@
 # necessary) and fed to cmp or diff.  The exit status from cmp
 # or diff is preserved.
 
-PATH="/usr/bin:/bin:$PATH"; export PATH
+PATH="/var/usr/bin:/usr/bin:/bin:$PATH"; export PATH
 prog=`echo $0 | sed 's|.*/||'`
 case "$prog" in
   *cmp) comp=${CMP-cmp}   ;;
--- src/bzgrep.orig
+++ src/bzgrep
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/var/usr/bin/sh
 
 # Bzgrep wrapped for bzip2, 
 # adapted from zgrep by Philippe Troin <phil@fifi.org> for Debian GNU/Linux.
@@ -6,7 +6,7 @@
 ## zgrep -- a wrapper around a grep program that decompresses files as needed
 ## Adapted from a version sent by Charles Levert <charles@comm.polymtl.ca>
 
-PATH="/usr/bin:$PATH"; export PATH
+PATH="/var/usr/bin:/usr/bin:$PATH"; export PATH
 
 prog=`echo $0 | sed 's|.*/||'`
 case "$prog" in
