--- src/print.c.orig
+++ src/print.c
@@ -1178,29 +1178,3 @@
 	else
 		(void)printf("%*s", v->width, "-");
 }
-
-void
-wq(KINFO *k, VARENT *ve)
-{
-	VAR *v;
-	struct proc_workqueueinfo wqinfo;
-	int len;
-	int ret;
-	uint32_t nthreads;
-
-	len = sizeof(wqinfo);
-	ret = proc_pidinfo(KI_PROC(k)->p_pid, PROC_PIDWORKQUEUEINFO, 0, &wqinfo, len);
-
-	v = ve->var;
-
-	if (len == ret && len == PROC_PIDWORKQUEUEINFO_SIZE) {
-		if (strcmp(v->name, "wqr") == 0)
-			nthreads = wqinfo.pwq_runthreads;
-		else if (strcmp(v->name, "wqb") == 0)
-			nthreads = wqinfo.pwq_blockedthreads;
-		else
-			nthreads = wqinfo.pwq_nthreads;
-		printf("%*d", v->width, nthreads);
-	} else
-		printf("%*s", v->width, "-");
-}
--- src/keyword.c.orig
+++ src/keyword.c
@@ -195,9 +195,6 @@
 	{"vsize", "", "vsz"},
 	{"vsz", "VSZ", NULL, 0, vsize, NULL, 8},
 	{"wchan", "WCHAN", NULL, LJUST, wchan, NULL, 6},
-	{"wq", "WQ", NULL, 0, wq, NULL, 2, 0, CHAR, NULL, 0},
-	{"wqb", "WQB", NULL, 0, wq, NULL, 2, 0, CHAR, NULL, 0},
-	{"wqr", "WQR", NULL, 0, wq, NULL, 2, 0, CHAR, NULL, 0},
 	{"xstat", "XSTAT", NULL, 0, pvar, NULL, 4, POFF(p_xstat), USHORT, "x"},
 	{""},
 };
