--- source/src/main.c.orig	2023-01-21 03:32:37.567629427 -0500
+++ source/src/main.c	2023-01-21 03:32:54.427552315 -0500
@@ -257,8 +257,6 @@
     if(exclude_addfile(arg)) die("Can't open %s: %s\n", arg, strerror(errno));
   } else if(OPT("--exclude-caches")) cachedir_tags = 1;
   else if(OPT("--include-caches")) cachedir_tags = 0;
-  else if(OPT("--exclude-kernfs")) exclude_kernfs = 1;
-  else if(OPT("--include-kernfs")) exclude_kernfs = 0;
   else if(OPT("--follow-firmlinks")) follow_firmlinks = 1;
   else if(OPT("--exclude-firmlinks")) follow_firmlinks = 0;
   else if(OPT("--confirm-quit")) confirm_quit = 1;
@@ -291,9 +289,6 @@
   printf("  -X, --exclude-from FILE    Exclude files that match any pattern in FILE\n");
   printf("  -L, --follow-symlinks      Follow symbolic links (excluding directories)\n");
   printf("  --exclude-caches           Exclude directories containing CACHEDIR.TAG\n");
-#if HAVE_LINUX_MAGIC_H && HAVE_SYS_STATFS_H && HAVE_STATFS
-  printf("  --exclude-kernfs           Exclude Linux pseudo filesystems (procfs,sysfs,cgroup,...)\n");
-#endif
 #if HAVE_SYS_ATTR_H && HAVE_GETATTRLIST && HAVE_DECL_ATTR_CMNEXT_NOFIRMLINKPATH
   printf("  --exclude-firmlinks        Exclude firmlinks on macOS\n");
 #endif
@@ -395,10 +390,6 @@
     else if(!arg_option()) die("Unknown option '%s'.\n", argparser_state.last);
   }
 
-#if !(HAVE_LINUX_MAGIC_H && HAVE_SYS_STATFS_H && HAVE_STATFS)
-  if(exclude_kernfs) die("The --exclude-kernfs flag is currently only supported on Linux.\n");
-#endif
-
   if(export) {
     if(dir_export_init(export)) die("Can't open %s: %s\n", export, strerror(errno));
     if(strcmp(export, "-") == 0) ncurses_tty = 1;
