--- src/Utilities/cmcurl/CMakeLists.txt.orig
+++ src/Utilities/cmcurl/CMakeLists.txt
@@ -929,12 +929,6 @@
   endif()
   list(APPEND CURL_LIBS "-framework CoreFoundation")
 
-  find_library(CORESERVICES_FRAMEWORK NAMES "CoreServices")
-  mark_as_advanced(CORESERVICES_FRAMEWORK)
-  if(NOT CORESERVICES_FRAMEWORK)
-    message(FATAL_ERROR "CoreServices framework not found")
-  endif()
-  list(APPEND CURL_LIBS "-framework CoreServices")
 endif()
 
 if(CURL_USE_OPENSSL)
--- src/Utilities/cmlibarchive/CMakeLists.txt.orig
+++ src/Utilities/cmlibarchive/CMakeLists.txt
@@ -2182,7 +2182,6 @@
 
 # We need CoreServices on Mac OS.
 IF(APPLE)
-  LIST(APPEND ADDITIONAL_LIBS "-framework CoreServices")
 ENDIF(APPLE)
 
 add_subdirectory(libarchive)
--- src/Source/CMakeLists.txt.orig
+++ src/Source/CMakeLists.txt
@@ -1047,7 +1047,9 @@
 # On Apple we need CoreFoundation and CoreServices
 if(APPLE)
   target_link_libraries(CMakeLib PUBLIC "-framework CoreFoundation")
+  if(HAVE_CoreServices)
   target_link_libraries(CMakeLib PUBLIC "-framework CoreServices")
+  endif()
 endif()
 
 if(WIN32 AND NOT UNIX)
