--- src/posix/gawkmisc.c.orig
+++ src/posix/gawkmisc.c
@@ -344,7 +344,7 @@
 			(void) unsetenv("GAWK_PMA_REINCARNATION");
 	}
 #endif
-#ifdef HAVE__NSGETEXECUTABLEPATH
+#if defined(HAVE__NSGETEXECUTABLEPATH) && !(defined(__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__) && __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ < 1050)
 	// This code is for macos
 	if (persist_file != NULL) {
 		const char *cp = getenv("GAWK_PMA_REINCARNATION");
